.github/workflows/build-test.yaml

```diff
@@ -3,66 +3,73 @@ on: [push]
+          path: ~/sonar/cache
           key: ${{ runner.os }}-sonar
           restore-keys: ${{ runner.os }}-sonar
       - name: Cache SonarCloud scanner
         id: cache-sonar-scanner
-        uses: actions/cache@v1
+        uses: actions/cache@v4
         with:
-          path: .\.sonar\scanner
+          path: ./.sonar/scanner
           key: ${{ runner.os }}-sonar-scanner
           restore-keys: ${{ runner.os }}-sonar-scanner
       - name: Build
         run: dotnet build --configuration Release
       - name: Install SonarCloud scanner
         if: steps.cache-sonar-scanner.outputs.cache-hit != 'true'
-        shell: powershell
         run: |
-          New-Item -Path .\.sonar\scanner -ItemType Directory
-          dotnet tool update dotnet-sonarscanner --tool-path .\.sonar\scanner
-      - name: Test & Analyze
+          mkdir -p ./.sonar/scanner
+          dotnet tool update dotnet-sonarscanner --tool-path ./.sonar/scanner
+      - name: Start SonarCloud scanner
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        shell: powershell
         run: |
-          .\.sonar\scanner\dotnet-sonarscanner begin `
-          /k:"Qowaiv_Qowaiv" `
-          /o:"qowaiv" `
-          /d:sonar.token="${{ secrets.SONAR_TOKEN }}" `
-          /d:sonar.host.url="https://sonarcloud.io" `
-          /d:sonar.cs.opencover.reportsPaths="**/coverage.**.opencover.xml" `
+          ./.sonar/scanner/dotnet-sonarscanner begin \
+          /k:"Qowaiv_Qowaiv" \
+          /o:"qowaiv" \
+          /d:sonar.token="${{ secrets.SONAR_TOKEN }}" \
+          /d:sonar.host.url="https://sonarcloud.io" \
+          /d:sonar.cs.opencover.reportsPaths="**/coverage.**.opencover.xml" \
           /d:sonar.cs.vstest.reportsPaths="**/*.trx"
-
-          dotnet test specs/Qowaiv.Specs --configuration Release `
-          --logger trx `
-          /p:CollectCoverage=true `
-          /p:ThresholdType=branch `
+      - name: Test
+        run: |
+          dotnet test specs/Qowaiv.Specs --configuration Release \
+          --logger "trx;LogFileName=test-results.trx" \
+          /p:CollectCoverage=true \
+          /p:ThresholdType=branch \
           /p:CoverletOutputFormat=opencover
-          
-          .\.sonar\scanner\dotnet-sonarscanner end /d:sonar.token="${{ secrets.SONAR_TOKEN }}"
+      - name: Stop SonarCloud scanner
+        if: success() || failure()
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: |
+          ./.sonar/scanner/dotnet-sonarscanner end \
+          /d:sonar.token="${{ secrets.SONAR_TOKEN }}"
+      - name: Upload test results
+        uses: actions/upload-artifact@v4
+        if: success() || failure()
```

Comment:
What is the reasoning about this? this seems always to be true, but I guess it is not that trivial?